### ISSUE TITLE
Efficio CPP has Leap Motion Frames!

### ISSUE THREAD

**Issue by jamesAbantech** [Abantech/EfficioCPP#1]
The title says it all
![image](https://cloud.githubusercontent.com/assets/10690276/17457854/69023ca2-5bd1-11e6-8c39-171dd831817b.png)

**Comment by GMelencio** [comment]
Hooray @jamesAbantech!! One device toward cross platform goodness! Keep at the near side while I take the far-side work especially since it would be great to have display on the Vive

**Comment by jamesAbantech** [comment]
You'll see that the Leap data can now be pulled from the frame.  This isn't our final model on how it will look but I am just experimenting with the concept of embedding types that aren't native to our DLL. C++ projects are wonky.

![image](https://cloud.githubusercontent.com/assets/10690276/17459304/2c979b2a-5c00-11e6-8513-d759c98a7437.png)

**Comment by jamesAbantech** [comment]
In order to get this project building, make a LEAP_SDK system environment variable to point to your Leap Motion SDK folder.

**Comment by theo-armour** [comment]
Looks cool. And I have not the faintest idea what it does or who it's for. ;-)

A nice read me file would be great.

Some examples:

* https://github.com/jbeder/yaml-cpp
* https://github.com/crossbario/autobahn-cpp

And access to many more:

https://github.com/fffaraz/awesome-cpp/blob/master/README.md

In most cases the Read is both the documentation and the primary marketing and promotions document. So spending to time and thought in creating a good readme is valuable and worth the effort.

**Comment by GMelencio** [comment]
@theo-armour: I undertand the need to document the thought process for
others who want to catch up, and we will in due time. This has been
discussed and docuumented at length in our meeting notes and James and I
have gone over it many times so we kno exactly what it is - maybe some
incentive to have short brainstorming meetings regularly especailly since
James will be travelling the globe ;)

Here's the TL;DR version:

Since device compatibility/API's least common denominator is C++, we are
rewriting the Efficio engine in C++. What about Java Script and all those
other goodies? Well fortunately, with C++ we can SWIG
<http://www.swig.org/> the
Efficio code to any language (C#, Python, even JavaScript) so long as the
core libs are written in C++. What James is showing is the start of the
Efficio C++ library which is now able to receive frames from the leap
device. once we can turn that into useful information (i.e. skeletal model)
we can port it out to the other applications.


@jamesAbantech - I finally got the project up and running but am not
getting any frames from the leap controller despite having creating a
LEAP_SDK environment variable and pointing it to the Leap sdk dir...
seescreenshot below:

[image: Inline image 1]


Greg Melencio
CEO and Founder
Abantech LLC
571-402-4688

**Comment by GMelencio** [comment]
@jamesAbantech

Just want to add here that I was not 100% correct - althiguh I dont see "i"
in the same way you do, in Engine.cpp, I DO see that I can get the current
Frames Per Second! Woohoo!!

[image: Inline image 1]

Greg Melencio
CEO and Founder
Abantech LLC
571-402-4688

**Comment by theo-armour** [comment]
@Abantech/core 

>  Well fortunately, with C++ we can SWIG

Swig looks cool.

I note that people seem to talk about emscripten a lot more.

https://www.google.com/trends/explore?date=all&q=emscripten,swig.org

>  I understand the need to document the thought process for others who want to catch up, and we will in due time.

If you don't welcome peeps in now, then you are no likely to ever do so.

In that case, better to go the closed source/proprietary route.

> {porting to other languages}

All doable. But I note that porting to client-side is its own specially little hell.  ;-)

***

But this pales in comparison to the ongoing need to define some useful products or potential benefits to peeps.

In other words. It's nice that Efficient is messaging. But can we find some cool messages to send?
